tests/test_maxcut.py

```diff
@@ -54,7 +54,26 @@ def test_maxcut_qaoa_obj_fixed_angles():
             gamma, beta, AR = get_fixed_gamma_beta(d, p, return_AR=True)
             for simulator in ["auto", "qiskit"]:
                 f = get_qaoa_maxcut_objective(N, p, G=G, parameterization="gamma beta", simulator=simulator)
-                assert -f(gamma, beta) / optimal_cut > AR
+                assert f(gamma, beta) / optimal_cut < AR
+
+
+def test_maxcut_qaoa_obj_fixed_angles_with_terms_and_precomputed_energies():
+    N = 10
+    for d, max_p in [(3, 11), (5, 4)]:
+        G = nx.random_regular_graph(d, N)
+        obj = partial(maxcut_obj, w=get_adjacency_matrix(G))
+        precomputed_energies = precompute_energies(obj, N)
+        optimal_cut, x = brute_force(obj, N, function_takes="bits")
+        for p in range(1, max_p + 1):
+            gamma, beta, AR = get_fixed_gamma_beta(d, p, return_AR=True)
+            for simulator in ["auto"]:
+                f1 = get_qaoa_maxcut_objective(N, p, G=G, parameterization="gamma beta", simulator=simulator)
+                f2 = get_qaoa_maxcut_objective(N, p, precomputed_cuts=precomputed_energies, parameterization="gamma beta", simulator=simulator)
+                e1 = f1(gamma, beta)
+                e2 = f2(gamma, beta)
+                assert e1 / optimal_cut < AR
```

Comment:
@rsln-s 
As you mentioned in previous commits, the desired behavior was
Energy: -7.98, Overlap: 0.85
Energy: -8.74, Overlap: 0.65
Energy: -9.24, Overlap: 0.46
Energy: -9.55, Overlap: 0.25
Energy: -9.76, Overlap: 0.12
Energy: -9.85, Overlap: 0.09
Energy: -9.87, Overlap: 0.08
Energy: -9.87, Overlap: 0.09
Energy: -9.85, Overlap: 0.10
Energy: -9.86, Overlap: 0.09
Energy: -9.90, Overlap: 0.06

Since the energy output is -ve f(gamma, beta) is -ve.

If you are ok with below behavior let me know I can make the changes
Energy: 7.98, Overlap: 0.85
Energy: 8.74, Overlap: 0.65
Energy: 9.24, Overlap: 0.46
Energy: 9.55, Overlap: 0.25
Energy: 9.76, Overlap: 0.12
Energy: 9.85, Overlap: 0.09
Energy: 9.87, Overlap: 0.08
Energy: 9.87, Overlap: 0.09
Energy: 9.85, Overlap: 0.10
Energy: 9.86, Overlap: 0.09
Energy: 9.90, Overlap: 0.06